player/lua/console.lua

```diff
@@ -1837,8 +1837,9 @@ mp.register_script_message('type', function(text, cursor_pos)
 end)
 
 mp.register_script_message('get-input', function (script_name, args)
-    if repl_active then
-        return
+    if repl_active and input_caller and script_name ~= input_caller then
+        mp.commandv('script-message-to', input_caller, 'input-event',
+                    'closed', utils.format_json({line, cursor}))
```

Comment:
This looks hacky.

Reply:
Better suggestions welcome.

Reply:
Have a stack with the state, and restore whatever was/is open after select ended.

Reply:
I meant the other way, to immediate go to user requested input, but go back to previous state, if requested (closed without select). i.e. allow nested menus in menu type situation.

Reply:
Isn't it strange to go back to the previous state when it was invoked by a completely different script doing something completely different? I think it should still send `closed` to different scripts like in this PR. Restoring the previous state from the same script is a separate functionality that can be added later.

Reply:
I think close without select or indeed another dedicated function could achieve that.

As for the rest, I don't think other scripts should be able to interrupt the selection that is being made, with implicit way. If it is invoked as "nested" select we can support that to provide "menu" like interface, but I don't see others nuking already open select. That's why I feel that it is hacky, because we allow it to open and send some implicit closed events, but only if script is different, it's weird.

Reply:
That's fair though I think one reasonable use case for it is to try out `script-binding select/...` bindings in the console, this was actually requested in https://github.com/mpv-player/mpv/pull/14087#issuecomment-2109128569 (and I misunderstood it as referring to the input queue at that time too).